### ISSUE TITLE
mlx.nn.losses.binary_cross_entropy should have class weights support

### ISSUE THREAD

**Issue by ubayram** [ml-explore/mlx-examples#175]
For training with imbalanced datasets, providing the weights for classes is one of the most common features I use when training neural networks. The mlx implementation of this loss doesn't have this feature implemented yet, so wanted to point that out.

> class mlx.nn.losses.binary_cross_entropy(logits: [array](https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.array.html#mlx.core.array), targets: [array](https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.core.array.html#mlx.core.array), reduction: [str](https://docs.python.org/3/library/stdtypes.html#str) = 'none')

**Comment by guptaaryan16** [comment]
hey @awni can I work on this? seems good to start on MLX

**Comment by awni** [comment]
That would be great!

**Comment by sujantkumarkv** [comment]
since this issue is fixed with the merged PR, @awni I would request to close this issue.
